cmd/samples/pso/workflow.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/cadence"
+	"go.uber.org/zap"
+
+	"github.com/pborman/uuid"
+	"go.uber.org/cadence/workflow"
+)
+
+type (
+	individual struct {
+		IndividualID string
+		Variable     []float64
+		Fitness      []float64
+	}
+)
+
+// ApplicationName is the task list for this sample
+const ApplicationName = "PSO"
+
+// HostID - Use a new uuid just for demo so we can run 2 host specific activity workers on same machine.
+// In real world case, you would use a hostname or ip address as HostID.
+var HostID = ApplicationName + "_" + uuid.New()
+
+var ActivityOptions = workflow.ActivityOptions{
+	ScheduleToStartTimeout: time.Second * 5,
+	StartToCloseTimeout:    time.Minute * 10,
+	HeartbeatTimeout:       time.Second * 2, // such a short timeout to make sample fail over very fast
+	RetryPolicy: &cadence.RetryPolicy{
+		InitialInterval:          time.Second,
+		BackoffCoefficient:       2.0,
+		MaximumInterval:          time.Minute,
+		ExpirationInterval:       time.Minute * 10,
+		MaximumAttempts:          5,
+		NonRetriableErrorReasons: []string{"bad-error"},
+	},
+}
+
+// This is registration process where you register all your workflow handlers.
+func init() {
+	//workflow.Register(FitnessEvaluationWorkflow)
+	workflow.Register(PSOWorkflow)
+	workflow.Register(PSOChildWorkflow)
+}
+
+//PSOWorkflow workflow decider
+func PSOWorkflow(ctx workflow.Context, functionName string) (err error) {
+	logger := workflow.GetLogger(ctx)
```

Comment:
consider adding queries to report current state of the simulation

Reply:
Great idea